src/components/EventForm/EventForm.tsx

```diff
@@ -0,0 +1,186 @@
+import { formatISO, format } from 'date-fns';
+
+import Alert from 'src/components/Alert';
+import Button from 'src/components/inputs/Button';
+import DateField from 'src/components/fields/DateField';
+import SelectField from 'src/components/fields/SelectField';
+import TextAreaField from 'src/components/fields/TextAreaField';
+import TextField from 'src/components/fields/TextField';
+import TimeField from 'src/components/fields/TimeField';
+import { EventFormWrapper, Form } from './styled';
+import { EVENT_OPTIONS } from './constants';
+
+export type EventDataInput = {
+  id?: string;
+  name?: string;
+  description?: string;
+  startTime?: Date;
+  endTime?: Date;
+  imageUrl?: string;
+  imageAlt?: string;
+  location?: string;
+  eventType?: string;
+};
+
+export type EventDataOutput = {
+  id?: string;
+  name: string;
+  description: string;
+  startTime: Date;
+  endTime: Date;
+  imageUrl?: string;
+  imageAlt?: string;
+  location?: string;
+  eventType: string;
+};
+
+type EventFormProps = {
+  headerTitle: string;
+  onSubmit: (data: EventDataOutput) => Promise<void>;
+} & EventDataInput;
+
+function EventForm({ onSubmit, ...rest }: EventFormProps) {
+  const [name, setName] = useState(rest.name || '');
+  const [imageUrl, setImageUrl] = useState(rest.imageUrl || '');
+  const [imageAlt, setImageAlt] = useState(rest.imageAlt || '');
+  const [date, setDate] = useState(
+    format(
+      new Date((rest.startTime || new Date(Date.now())).toDateString()) ||
+        Date.now(),
+      'yyyy-MM-dd'
+    )
+  );
+  const [startTime, setStartTime] = useState(
+    format(rest.startTime || Date.now(), 'HH:mm')
+  );
+  const [endTime, setEndTime] = useState(
+    format(rest.endTime || Date.now(), 'HH:mm')
+  );
+  const [description, setDescription] = useState(rest.description || '');
+  const [eventType, setEventType] = useState(rest.eventType || '');
+  const [location, setLocation] = useState(rest.location || '');
+
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  const [hasSubmitted, setHasSubmitted] = useState(false);
+  const [error, setError] = useState<string | null>(null);
+
+  const isFormValid = name && description && eventType;
+  const isSubmitDisabled = !isFormValid || isSubmitting;
+
+  const handleSubmit = useCallback(async () => {
+    setIsSubmitting(true);
+    setError(null);
+
+    try {
+      await onSubmit({
+        id: rest.id,
+        name,
+        description,
+        startTime: new Date(date + ' ' + startTime),
+        endTime: new Date(date + ' ' + endTime),
+        imageUrl,
+        imageAlt,
+        location,
+        eventType,
+      });
+    } catch (error) {
+      setError(error.message);
+    } finally {
+      setIsSubmitting(false);
+    }
+    setHasSubmitted(true);
+  }, [
+    name,
+    description,
+    date,
+    startTime,
+    endTime,
+    imageUrl,
+    imageAlt,
+    location,
+    eventType,
+    setIsSubmitting,
+    setError,
+    onSubmit,
+    rest,
```

Comment:
I think you can replace `rest` with `rest.id` here. I don't really think it will have any performance implications here, but I generally try to be as explicit as possible when specifying hook dependencies to avoid unnecessary re-renders.
```suggestion
    rest.id,
```